### PR TITLE
boards: st: st25dv_mb1283_disco: add mb1283 board

### PR DESCRIPTION
**Pull request by everedero**
ST25DV discovery kit, including an LCD color screen. ST25DV support not included.

### CODE REVIEW

File: boards/st/st25dv_mb1283_disco/Kconfig.st25dv_mb1283_disco [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# ST25DV DISCOVERY board configuration
+
+# Copyright (c) 2024 Eve Redero
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_ST25DV_MB1283_DISCO
+	bool "ST25DV discovery board with MB1283"
```
Comment (erwango) [inline boards/st/st25dv_mb1283_disco/Kconfig.st25dv_mb1283_disco:7]: Remove `bool "...."`

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM, thanks!

**Comment by ajarmouni-st** [comment]
Hi @everedero, can you please test samples/modules/lvgl/demos & samples/drivers/display as well, if you haven't already, & confirm that they work correctly? Thanks!

**Comment by everedero** [comment]
- samples/drivers/display work out of the box without any overlay, should I add a configuration file or something?

- I added overlay and conf files to lvgl samples, music player works despite being very slow. 

- Also, I will open a ticket because the "zephyr,lvgl-keypad-input" driver makes the lvgl sample compilation fail. In the meantime, I will leave it out of the overlay file in lvgl samples, if that's OK with you - afaik these samples do not use lvgl-keypad-input.

**Review by erwango — DISMISSED** [review]
Approved but ideally you should split boards and samples changes in 2 separate commits.

**Comment by ajarmouni-st** [comment]
> should I add a configuration file or something?

No.

> I added overlay and conf files to lvgl samples, music player works despite being very slow.

Great, thanks!

>  will leave it out of the overlay file in lvgl samples, if that's OK with you

Yes no problem

**Review by ajarmouni-st — DISMISSED** [review]

File: boards/st/st25dv_mb1283_disco/st25dv_mb1283_disco.dts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+		reset-gpios = <&gpioc 1 GPIO_ACTIVE_HIGH>;
+		dc-gpios = <&gpioc 0 GPIO_ACTIVE_HIGH>;
+		spi-dev = <&spi2>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		write-only;
+
+		ili9341: ili9341@0 {
+			compatible = "ilitek,ili9341";
+			mipi-max-frequency = <20000000>;
```
Comment (GeorgeCGV) [inline boards/st/st25dv_mb1283_disco/st25dv_mb1283_disco.dts:87]:
```suggestion
			mipi-max-frequency = <DT_FREQ_M(20)>;
```

File: boards/st/st25dv_mb1283_disco/st25dv_mb1283_disco.dts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,display = &ili9341;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		orange_led: led_1 {
+			gpios = <&gpioc 13 GPIO_ACTIVE_HIGH>;
+			label = "Orange LD1";
```
Comment (GeorgeCGV) [inline boards/st/st25dv_mb1283_disco/st25dv_mb1283_disco.dts:29]: `label` is deprecated for quite some time

File: boards/st/st25dv_mb1283_disco/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# Copyright (c) 2024 Eve Redero
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_ST25DV_MB1283_DISCO
+
+config SPI_STM32_INTERRUPT
+	default y
+	depends on SPI
+
+endif # BOARD_STM25DV_MB1283_DISCO
```
Comment (kartben) [inline boards/st/st25dv_mb1283_disco/Kconfig.defconfig:12]:
```suggestion
endif # BOARD_ST25DV_MB1283_DISCO
```

File: boards/st/st25dv_mb1283_disco/docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+- LD2 : PC4 (Yellow led)
+- GPIO : PE15/PE7 (Available on external ST25DV connector J6)
+- GPIO : PE14 (Available on J5 connector)
+
+Serial Port
+===========
+
+The Zephyr console output and shell are assigned to UART6, which is connected to the
+onboard ST-LINK/V2 Virtual COM port interface.
+which connected to the onboard ST-LINK/V2 Virtual COM port interface.
```
Comment (kartben) [inline boards/st/st25dv_mb1283_disco/docs/index.rst:81]: copy-paste issue here, or something?

File: boards/st/st25dv_mb1283_disco/docs/index.rst [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+--------------------------------------------
+
+First, connect the STM25DV Discovery kit to your host computer using
+the USB port to prepare it for flashing. Then build and flash your application.
+
+Here is an example for the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: stm25dv_disco
```
Comment (kartben) [inline boards/st/st25dv_mb1283_disco/docs/index.rst:109]: this doesn't look like the right name :) In fact, there are many "STM25DV" instances in your patch anyway, that probably call for a global search and replace

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by GeorgeCGV — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**